src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -69,14 +90,50 @@ class PeerManagerActor(
     peer
   }
 
+  def getPeers(): Future[Peers] = {
+    implicit val timeout = Timeout(2.seconds)
+
+    Future.traverse(peers.values) { peer =>
+      (peer.ref ? PeerActor.GetStatus)
+        .mapTo[PeerActor.StatusResponse]
+        .map(sr => (peer, sr.status))
+    }.map(r => Peers.apply(r.toMap))
+  }
+
+  def tryDiscardPeersToFreeUpLimit(): Future[Unit] = {
+    getPeers() flatMap { peers =>
+      val peersToPotentiallyDisconnect = peers.peers.filter {
+        case (_, Handshaking(numRetries)) if numRetries > 0 => true /* still handshaking and retried at least once */
+        case (_, PeerActor.Status.Disconnected) => true /* already disconnected */
+        case _ => false
+      }
+
+      if (peers.peers.size - peersToPotentiallyDisconnect.size >= peerConfiguration.maxPeers) {
+        Future.failed(new RuntimeException("Too many peers"))
```

Comment:
I'm wondering, should we try to connect to at least some of the handshaked peers here?
For example I tried having `max-peers=2` and I got the following logs:
```
17:51:15.337 i.i.e.network.PeerManagerActor - Maximum number of connected peers reached. Not connecting to enode://18a551bee469c2e02de660ab01dede06503c986f6b8520cb5a65ad122df88b17b285e3fef09a40a0d44f99e014f8616cf1ebc2e094f96c6e09e2f390f5d34857@47.90.36.129:30303
17:51:15.339 i.i.e.network.PeerManagerActor - Maximum number of connected peers reached. Not connecting to enode://98b863da48ab8bef2339a823d552f3619fd8e892425ae40c6812c6f7e4a0afb4f9591b012183e89a63bb01c5085d0e96aa5f0812652335fb0ac946d6aaf15881@118.178.57.121:30303
17:51:15.363 i.i.e.blockchain.sync.SyncController - Block: 66176/3484815. Peers: 1/1 (0 blacklisted). State: 97425/108279 known nodes.
```
Shouldn't we in this case connect to one of the two nodes as the number of connected nodes is less than `max-peers`?